### PR TITLE
Upgrading to 0.12

### PR DESCRIPTION
**Pull request by endrec**
This PR addresses compatibility issues with terraform 0.12.x

### CODE REVIEW

File: spotinst_ocean_cnvrg/main.tf [hunk 1]
```diff
@@ -34,123 +35,128 @@ resource "aws_security_group" "all_worker_mgmt" {
...
   enable_nat_gateway = true
   single_nat_gateway = true
-  tags               = "${merge(local.tags, map("kubernetes.io/cluster/${local.cluster_name}", "shared"))}"
+  tags = merge(
+    local.tags,
+    {
+      "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    },
+  )
 }
```
Comment (seanmcrw) [inline spotinst_ocean_cnvrg/main.tf:53]: We don't need to define ```module "vpc"``` here. We just need to use the data remote state resource to reference the vpc we already have. 

```
data "terraform_remote_state" "eks-vpc" {
  backend = "s3"
  config = {
    bucket = "rw-terraformstate-ops"
    region = "eu-west-1"
    key    = "eks-vpc/terraform.tfstate"
  }
}
```
then

```
private_subnets    = data.terraform_remote_state.eks-vpc.outputs.private_subnets
```

**Review by seanmcrw — COMMENTED** [review]
I think we should stay away from nested modules ```module "eks"```
i think I see this as x2 modules. One for EKS and one for Spotinst. EKS would be our reworking of the reg version without the assumption that we want to use ASG's.

## Assumptions

* You want to create an EKS cluster and an autoscaling group of workers for the cluster.
* You want these resources to exist within security groups that allow communication and coordination. These can be user provided or created within the module.
* You've created a Virtual Private Cloud (VPC) and subnets where you intend to put the EKS resources. The VPC satisfies [EKS requirements](https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html).
* If `manage_aws_auth = true`, it's required that both [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl) (>=1.10) and [`aws-iam-authenticator`](https://github.com/kubernetes-sigs/aws-iam-authenticator#4-set-up-kubectl-to-use-authentication-tokens-provided-by-aws-iam-authenticator-for-kubernetes) are installed and on your shell's PATH.

The Spotinst version would be our reworking of the reg version without the nested mods EKS + EKS. This way we can manage spotinst resources seperate from eks resources.

**Comment by endrec** [comment]
This is supposed to be an update of the spotinst supplied terraform module, so if we would like to stay away from nested modules, we need to roll our own modules, and we need to stay away from this module entirely.

**Comment by seanmcrw** [comment]
We can appropriate elements of the spotinst mod, no need to start from scratch.

**Pull request closed without merging**